text/0507-subnets.md

```diff
@@ -0,0 +1,402 @@
+  mapPublicIpOnLaunch: false, // default: true for public subnets
+});
+
+// The following is true
+vpc.publicSubnets.includes(subnet);
+
+// As is
+vpc.selectSubnets({subnetType: ec2.SubnetType.PUBLIC}).includes(subnet);
+```
+
+To create a route table that sends DynamoDB and S3 traffic through VPC
+endpoints:
+
+```ts
+const routeTable = new RouteTable(this, 'routeTable', {
+  routes: [
+    Route.DYNAMODB_ENDPOINT,
+    Route.S3_ENDPOINT
+  ],
+});
+```
+
+You can use a public NAT gateway to enable instances in a private subnet to
+send outbound traffic to the internet, while preventing the internet from
+establishing connections to the instances:
+
+```ts
+const elasticIp = new ElasticIp({
+  domain: Domain.VPC
+
+  // Other properties, such as networkBorderGroup and publicIpv4Pool, 
+  // are also available. Omitted here for brevity.
+});
+
+const natGateway = publicSubnet.addNatGateway('natgw', {elasticIp});
+
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.to({
+      destination: '0.0.0.0/0',
+
+      // targets must implement the IRouter interface
+      target: natGateway,
+    }),
+  ],
+});
+
+const privateSubnet = vpc.addSubnet('privateSubnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a',
+  routeTable,
+});
+```
+
+You can also create the same kind of routing pattern, but with a NAT instance
+instead:
+
+```ts
+const natInstance = publicSubnet.addNatInstance('natinst', {
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.MICRO),
+  machineImage: ec2.MachineImage.latestAmazonLinux({
+    generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2,
+  }),
+
+  // Other properties ommitted
+});
+
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.to({
+      destination: '0.0.0.0/0',
+
+      // Instance also implements IRouter
+      target: natInstance,
+    }),
+  ],
+});
+
+const privateSubnet = vpc.addSubnet('privateSubnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a',
+  routeTable,
+});
+```
+
+For IPv6 traffic, you create this pattern by using an egress-only internet
+gateway:
+
+```ts
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.local('10.0.0.0/16'),
+    Route.local('2001:db8:1234:1a00:/64'),
+
+    // The CIDR provided here must be for IPv6
+    Route.toEgressOnlyInternetGateway('::/0'),
+  ],
+});
+```
+
+To route traffic to a VPN Gateway, you can either explicitly add a route to the
+route table or you can enable route propagation:
+
+```ts
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.local('10.0.0.0/16'),
+
+    // Static route to a VPN Gateway. Takes priority over propagated ones
+    // Causes VPN gateway to be enabled in the VPC
+    Route.toVpnGateway('172.31.0.0/24'),
+  ],
+
+  // To make VPN Gateway routes propagate to this route table
+  enableVpnGatewayRoutePropagation: true, // default: false
```

Comment:
Is either of these fine or do we need both? The text says either but the code comments don't make it clear.

Reply:
Yeah, the text is wrong. You can have both, with the static one taking precedence.